src/sagas/producer.saga.js

```diff
@@ -0,0 +1,32 @@
+import { put, call, takeEvery } from 'redux-saga/effects';
+import {  producerApiData } from "../actions/index";
+import { REQUEST_API_DATA } from '../constants/action-types';
+
+const fetchApiData = async ()=>{
```

Comment:
this function should be agnostic of the url that it fetches meaning that it should take a url as a parameter